core/src/main/java/org/axonframework/config/EventHandlingConfiguration.java

```diff
@@ -451,34 +523,41 @@ public void initialize(Configuration config) {
             assignments.computeIfAbsent(processor, k -> new ArrayList<>()).add(handler);
         });
 
+        onInit.forEach(h -> h.accept(config));
```

Comment:
Not sure what the `onInit` call does...do you mind explaining this?

Reply:
`EventProcessorRegistry` is component withing configuration which is accessible at the moment when the configuration is given to `EventHandlingConfiguration` - `initialize()` method. Take a look at `EventHandlingConfiguration#registerSubscribingEventProcessor`, it needs `EventProcessorRegistry` in order to register the subscribing event processor, but it is not accessible at that moment. `onInit` is a list of consumers (of `Configuration`) which registers some intentions that are not possible until configuration is available to us. Hence, `onInit` handlers are invoked during initialization of `EventHandlingConfiguration`. I hope it is a bit clearer now :)

Reply:
Great, that makes sense :-)
It is what I deduced as well, but wanted to make sure I wasn't making any wrong assumption here, so thanks for that!